### DISCUSSION TITLE
Running longhorn on kind or miniKube

### DISCUSSION THREAD

**Discussion by orenzp** [longhorn/longhorn#2702] [Q&A] [answered]
Hi,
Is it possible to run longhorn on kind or minikube for testing CI  automation?

Thanks in advance,

**Comment by jenting** [comment, marked as answer]
Longhorn depends on [iscsi](https://longhorn.io/docs/1.1.1/deploy/install/#installation-requirements)
KIND and minikube are in container solution and are good for local development.

~However, two problems there, `iscsi` and `mount propagation` can't run successfully inside the KIND and minikube environment.~
However, the problem is `iscsi` can't run successfully inside the KIND and minikube environment.

**Reply by orenzp** [reply]
Thanks For the detailed answer @jenting

**Reply by kconde2** [reply]
Hello I have the same problem as described here https://github.com/longhorn/longhorn/issues/3643.

Do you think there are other alternatives to set up a local environment with longhorn?

**Reply by kconde2** [reply]
Can we use k3d that is k3s lightweight wrapper ?

Maybe I will test it to see.

Thanks in advance

**Reply by jenting** [reply]
k3d is not suitable for Longhorn as well since it's also a in container solution.

**Reply by innobead** [reply]
For testing purposes locally, have u tried https://github.com/innobead/kubefire

**Reply by innobead** [reply]
btw, if running minikube w/ virtual machine managers instead of container engine like docker, probably should work.

**Reply by kconde2** [reply]
Thanks I will try

**Reply by BenTheElder** [reply]
FWIW mount propagation actually can work in KIND at least (kubernetes has tests for this), and you can configure it on additional mounts from the host (though, not from say macOS host mounts or Windows host mounts), https://kind.sigs.k8s.io/docs/user/configuration/#extra-mounts).

iscsi may be tricky though https://www.docker.com/blog/road-to-containing-iscsi/
